### PR TITLE
[serve] Improve single-app and multi-app differentiation

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

- Disallow empty string application names in `ServeDeploySchema`
- Disallow specifying an app name if user is directly deploying a single-app config `ServeApplicationSchema`
  - Otherwise, `GET /api/serve/deployments/status` and `GET /api/serve/deployments/config` would not get the correct status and config. We could try to fix it, but there's no reason to allow users to specify the name of their application, so it's better to disallow it.
- Add a field `deploy_mode` in `ServeInstanceDetails` so that we can automatically detect what config(s) to show and ***REMOVE `--multi-app` flag in `serve config`***
- Remove trailing space for `serve config` and `serve status`

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zcin** [comment]
Please see updated description + code.

If we are OK with adding a field `deploy_mode` in `ServeInstanceDetails`, we can track whether a single-app or multi-app config has been deployed in the backend, and thus automatically decide what to show for `serve config`. So, I've removed the `--multi-app` flag in `serve config`.

File: dashboard/modules/serve/serve_agent.py [hunk 1]
```diff
@@ -166,6 +166,16 @@ async def put_all_deployments(self, req: Request) -> Response:
                 text=repr(e),
             )
 
+        if "name" in config.dict(exclude_unset=True):
+            return Response(
```
[the remark this answers is not in the document]
Comment (zcin) [reply dashboard/modules/serve/serve_agent.py]: Added!

**Review by sihanwang41 — APPROVED** [review]
please add test cases for `name` set in  `ServeApplicationSchema`.  Make sure the exception can be raised in serve_agent and deploy_apps.

**Comment by zcin** [comment]
> please add test cases for `name` set in `ServeApplicationSchema`. Make sure the exception can be raised in serve_agent and deploy_apps.

Added! @edoakes ping for review/merge

**Review by edoakes — APPROVED** [review]

**Pull request merged**